src/pages/BrowserPage.qml

```diff
@@ -40,7 +40,7 @@ Page {
     }
 
     function bringToForeground(window) {
-        if (!Qt.application.active && window) {
+        if ((webView.visibility < QuickWindow.Window.Maximized) && window) {
```

Comment:
I don't think that QWindow::AutomaticVisibility could cause issue here.

Reply:
As small note conceptually raise() should be valid operation with everything else than fullscreen. Or maybe even that if considering app being fullscreen and window app on top of it. Method here doesn't look like getting called too much so could even remove window state checks, though doesn't matter much.